### PR TITLE
CVE-2016-4436

### PR DESCRIPTION
**Pull request by JosefLenz**
History Project round 1

### CODE REVIEW

**Comment by JMdoubleU** [comment]
Can you double-check the report date? It looks like the first fix commit was 2016-04-19 (before the currently listed report date). This also applies to the discover date.
Additionally, from [the security bulletin for this issue](https://cwiki.apache.org/confluence/pages/viewpreviousversions.action?pageId=63930371), it looks like the announcement date is 2016-06-01.

It might be helpful to generalize the description a bit, as it's not quite clear what "Struts Action Mapper" is without knowledge of Struts. 

It's interesting that the impact of this is seemingly unknown (per the CVE and security bulletin). Are you able to find any sources that elaborate on the impact of this?

**Comment by JosefLenz** [comment]
The dates from varying sources were inconsistent, I thought i was able to sort them out, but ill go back through again. There want much explanation I could find either through code comments or commits as to what the action mapper's purpose was..

File: cves/CVE-2016-4436.yml [hunk 1]
```diff
@@ -76,12 +80,12 @@ incomplete_fix_instructions: |
   this CVE below.
 previous_fixes:
   - commit:
-    note:
+    note: The above fixes fixed the vulnerability
```
Comment (mesternefeld) [inline cves/CVE-2016-4436.yml:84]: You can keep this one note and get rid of the other comments below.

Comment (JosefLenz) [reply cves/CVE-2016-4436.yml:84]: good point thanks

File: cves/CVE-2016-4436.yml [hunk 2]
```diff
@@ -142,7 +148,10 @@ interesting_commits:
     emerging themes?
     If there are no interesting commits, demonstrate that you completed this section
     by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: |
+    the vulnerability was introduced nearly 10 years before it was discovered
+    there is a lot that happened inbetween the two commits, there are many interesting
```
Comment (cxd3796) [inline cves/CVE-2016-4436.yml:153]: Is there anything in between the two commits that may have led the developers to go down a path that would create the vulnerability?

File: cves/CVE-2016-4436.yml [hunk 3]
```diff
@@ -174,8 +186,8 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: The action mapper trused the input it received to be valid.
```
Comment (cxd3796) [inline cves/CVE-2016-4436.yml:190]: trusted*

File: cves/CVE-2016-4436.yml [hunk 4]
```diff
@@ -205,4 +217,9 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    It was assumed that no invalid action types would reach the action mapper.
+    It is a design mistake violating defense in depth to assume that invalid data
+    wouldnt reach an internal part of the system. The problem was fixed with a white list
```
Comment (cxd3796) [inline cves/CVE-2016-4436.yml:223]: wouldn't*

File: cves/CVE-2016-4436.yml [hunk 5]
```diff
@@ -205,4 +217,9 @@ mistakes:
...
+    checking that all action types being mapped match existing action types. This
+    prevents any incorrect input that could cause unexpeted behavior to the action mapper.
```
Comment (cxd3796) [inline cves/CVE-2016-4436.yml:225]: unexpected*

**Pull request closed without merging**